### PR TITLE
Call finalize in Series.str

### PR DESCRIPTION
**Pull request by TomAugspurger**
xref #28283

### CODE REVIEW

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -473,7 +473,8 @@ Other
 
 - Bug in :meth:`DataFrame.replace` and :meth:`Series.replace` incorrectly raising ``AssertionError`` instead of ``ValueError`` when invalid parameter combinations are passed (:issue:`36045`)
 - Bug in :meth:`DataFrame.replace` and :meth:`Series.replace` with numeric values and string ``to_replace`` (:issue:`34789`)
-- Fixed metadata propagation in the :class:`Series.dt` accessor (:issue:`28283`)
+- Fixed metadata propagation in the :class:`Series.dt` and :class:`Series.str` accessors (:issue:`28283`)
+- Bug in :meth:`Series.transform` would give incorrect results or raise when the argument ``func`` was dictionary (:issue:`35811`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:477]: is this an extra note?

**Review by jreback — CHANGES_REQUESTED** [review]
whatsnew doc note, otherwise lgtm.

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v1.2.0.rst:477]: I messed up fixing the merge conflict.
```suggestion
```

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**